### ISSUE TITLE
[BUG] prepublish gets called without first installing dev dependencies for file:/path dependencies.

### ISSUE THREAD

**Issue by trusktr** [npm/npm#9719]
For example, suppose we have package A that depends on package B, and package B has a prepublish script that depends on things installed from package B's dev dependencies.

Running `npm install` in package A will fail because those dev dependencies that package B needs for it's prepublish script are missing.

**Comment by KenanY** [comment]
Sounds like https://github.com/npm/npm/issues/9707.

**Comment by trusktr** [comment]
Yeah, seems like the same issue.

**Comment by snopeks** [comment]
Hello @trusktr I'm going to close this as a duplicate of #9707. If this turns out to be something different let me know and I'll be happy to reopen. Thanks!

**Comment by trusktr** [comment]
@KenanY ​@snopeks, it seems like the same problem, but his description was a little hard to follow.

Basically, the problem is this:

Package A depends on B. Package B is listed inside of package A as a local file-system dependency, so that package A has this in `package.json`:

```js
  "name": "A",
  "dependencies": {
    "B": "file:/path/to/package/B"
  }
```

When running `npm prepublish` on package A, the `prepublish` script of package B does not get executed.

I believe it is important for the prepublish script of package B (located in the local filesystem) to be published in order for it to be in it's usable state for package A (which is what `prepublish` is intended for as mentioned in #9707).
